docs/wave/aplp.rst

```diff
@@ -0,0 +1,210 @@
+.. default-role:: code
+
+All Pairs Longest Paths for Software Pipelining
+=========================================================
+
+This document explains the Rust library designed for All-Pairs Longest Path (APLP) computation, specifically tailored for software pipelining. The library consists of two main modules: `prune.rs` for optimizing path representations and `lib.rs` for the core APLP logic and Python FFI (Foreign Function Interface) using PyO3.
+
+The "paths" are represented as pairs `(delay, iter_diff)`, which define a line :math:`L(S) = \text{delay} - \text{iter_diff} \cdot S`, where :math:`S` is the symbolic initiation interval. The goal of pruning is to find the upper envelope of these lines.
```

Comment:
Just curious, is envelope a standard term? Just double checking, the envelope here you mean the hull boundaries right?

Reply:
I think kind of standard, but yes the upper envelope here is referring to the convex hull.